src/schema.rs

```diff
@@ -287,9 +317,27 @@ impl Schema {
         }
     }
 
+    fn parse_named(complex: &Map<String, Value>, context: &mut SchemaParseContext) -> Result<Self, Error> {
+        let name = Name::parse(complex)?;
+
+        let schema = match complex.get("type") {
+            Some(&Value::String(ref t)) => match t.as_str() {
+                "record" => Schema::parse_record(complex, context),
+                "enum" => Schema::parse_enum(complex),
+                "fixed" => Schema::parse_fixed(complex),
+                _ => panic!("parse_named got wrong type"),
```

Comment:
can we use `unreachable!()` here? It's used in other places in the codebase to indicate a behaviour that cannot happen 🙂